tests/MockApPipe.yaml

```diff
@@ -50,16 +51,14 @@ contracts:
   - imageDifference.doSkySources == transformDiaSrcCat.doRemoveSkySources
   # Inputs and outputs must match
   # Use of ConnectionsClass for templated fields is a workaround for DM-30210
-  - retrieveTemplate.connections.ConnectionsClass(config=retrieveTemplate).outputExposure.name ==
-      imageDifference.connections.ConnectionsClass(config=imageDifference).inputTemplate.name
```

Comment:
Why is this contract invalid? [The real pipeline](https://github.com/lsst/ap_pipe/blob/main/pipelines/ApPipe.yaml#L60-L61) uses it.

Reply:
There is no `retrieveTemplate` in the mock pipeline.

Reply:
So, to be fixed when we swap out `imageDifference`?